documents/extension/funding/service.go

```diff
@@ -264,4 +274,44 @@ func (s service) DeriveFundingResponse(model documents.Model, fundingID string)
 		Header: h,
 		Data:   data,
 	}, nil
+
+}
+
+// DeriveFundingListResponse returns a funding list in client format
+func (s service) DeriveFundingListResponse(model documents.Model) (*clientfundingpb.FundingListResponse, error) {
+	response := new(clientfundingpb.FundingListResponse)
+
+	h, err := documents.DeriveResponseHeader(s.tokenRegFinder(), model)
+	if err != nil {
+		return nil, errors.New("failed to derive response: %v", err)
+	}
+	response.Header = h
+
+	fl, err := documents.AttrKeyFromLabel(fundingLabel)
+	if err != nil {
+		return nil, err
+	}
+
+	if !model.AttributeExists(fl) {
+		return response, nil
+	}
+
+	lastIdx, err := getFundingsLatestIDX(model)
+	if err != nil {
+		return nil, err
+	}
+
+	i, err := documents.NewInt256("0")
+	if err != nil {
+		return nil, err
+	}
+
+	for ; i.Cmp(lastIdx) != 1; i.Inc() {
```

Comment:
is this valid? I thought i.Inc() returns error

Reply:
True.